src/sdk/accountspace_nb.js

```diff
@@ -0,0 +1,261 @@
+
+
+//const dummy_region = 'us-west-2';
+//const dummy_service_name = 's3';
+/* 
+    TODO: DISCUSS: 
+    1. IAM API only for account created using IAM API and OBC accounts not from admin, support, 
+       operator and account created using noobaa.
+    2. Do we need to have two access keys
+    3. get_access_key_last_used() API call could return dummy values?
+*/
+
+/**
+ * @implements {nb.AccountSpace}
+ */
+class AccountSpaceNB {
+    /**
+     * @param {{
+     *      rpc_client: nb.APIClient;
+     *      internal_rpc_client: nb.APIClient;
+     *      stats?: import('./endpoint_stats_collector').EndpointStatsCollector;
+     * }} params
+     */
+    constructor({ rpc_client, internal_rpc_client, stats }) {
+        this.rpc_client = rpc_client;
+        this.internal_rpc_client = internal_rpc_client;
+        this.stats = stats;
+    }
+
+    //////////////////////
+    // ACCOUNT METHODS  //
+    //////////////////////
+
+    async create_user(params, account_sdk) {
+
+        const action = IAM_ACTIONS.CREATE_USER;
+        //const requesting_account = account_sdk.requesting_account;
+        //const root_account = _.find(system_store.data.accounts, account => account.name.unwrap() === requesting_account.name.unwrap());
+        const requesting_account = system_store.get_account_by_email(account_sdk.requesting_account.email);
+        account_util._check_if_requesting_account_is_root_account(action, requesting_account,
+                { username: params.username, path: params.iam_path });
+        account_util._check_username_already_exists(action, params, requesting_account);
+        const iam_arn = iam_utils.create_arn_for_user(requesting_account._id.toString(), params.username, params.iam_path);
+        const account_name = new SensitiveString(`${params.username}:${requesting_account.name.unwrap()}`);
+        const req = {
+            rpc_params: {
+                name: account_name,
+                email: account_name,
+                has_login: false,
+                s3_access: true,
+                allow_bucket_creation: true,
+                owner: requesting_account._id.toString(),
+                is_iam: true,
+                iam_arn: iam_arn,
+                iam_path: params.iam_path,
+                role: 'iam_user',
+
+                // TODO: default_resource remove
+                default_resource: 'noobaa-default-backing-store',
+            },
+            account: requesting_account,
+        };
+        // CORE CHANGES PENDING - START
+        const iam_account = await account_util.create_account(req);
+        // CORE CHANGES PENDING - END
+
+        // TODO : Clean account cache
+        // TODO : Send Event
+        return {
+            // TODO : PATH Missing
+            iam_path: requesting_account.iam_path || IAM_DEFAULT_PATH,
+            username: params.username,
+            user_id: iam_account.id,
+            arn: iam_arn,
+            create_date: iam_account.create_date,
```

Comment:
I think `create_date` should be also saved, since we need it in `GetUser` and `ListUsers` APIs.
I will comment this also in the schema file.